enhancements/microshift/default-csi-plugin.md

```diff
@@ -0,0 +1,668 @@
+different. Include the reason for each choice in the design and
+implementation that is proposed here, and expand on reasons for not
+choosing alternatives in the Alternatives section at the end of the
+document.
+--->
+
+### Workflow Description
+
+<!---
+Explain how the user will use the feature. Be detailed and explicit.
+Describe all of the actors, their roles, and the APIs or interfaces
+involved. Define a starting state and then list the steps that the
+user would need to go through to trigger the feature described in the
+enhancement. Optionally add a
+[mermaid](https://github.com/mermaid-js/mermaid#readme) sequence
+diagram.
+
+Use sub-sections to explain variations, such as for error handling,
+failure recovery, or alternative outcomes.
+
+For example:
+
+**cluster creator** is a human user responsible for deploying a
+cluster.
+
+**application administrator** is a human user responsible for
+deploying an application in a cluster.
+
+1. The cluster creator sits down at their keyboard...
+2. ...
+3. The cluster creator sees that their cluster is ready to receive
+   applications, and gives the application administrator their
+   credentials.
+--->
+
+Device Owner: A human user with privileged access on the RHEL host and cluster-admin access in the MicroShift cluster.
+
+**Start MicroShift with default ODF-LVM configuration**
+
+> _Assuming a [RHEL 8.x](https://github.com/openshift/microshift/blob/main/docs/devenv_rhel8_auto.md) or 
+> [RHEL for Edge](https://github.com/openshift/microshift/blob/main/docs/rhel4edge_iso.md) host, ..._
+
+1. The device owner will connect to the device terminal.
+1. If the MicroShift service is not running, the device owner will start the service via `systemctl enable --now microshift`.
+1. The device owner will verify a stable deployment (see: **Validate-ODF-LVM function**)
+
+**Validate ODF-LVM function**
+1. The device owner will observe the cluster to confirm ODF-LVM pods are running with `oc get pods -n openshift-storage`.
+
+        [root@ushift ~]# oc get pods -n openshift-storage
+        NAME                                  READY   STATUS    RESTARTS        AGE
+        topolvm-controller-8479455f95-tdv7f   4/4     Running   0               30m
+        topolvm-node-clv2z                    4/4     Running   0               30m
+
+1. The device owner will create a simple Pod and PersistentVolumeClaim to test-drive the plugin with:
+```shell
+cat <<'EOF' | oc apply -f -
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: my-lv-pvc
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1G
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-pod
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    command: ["/usr/bin/sh", "-c"]
+    args: ["sleep", "1h"]
+    volumeMounts:
+    - mountPath: /mnt
+      name: my-volume
+  volumes:
+    - name: my-volume
+      persistentVolumeClaim:
+        claimName: my-lv-pvc
+EOF
+```
```

Comment:
Can we describe a workflow for the developer task of updating MicroShift to a new version of ODF-LVM, yet, or should that come later?

Reply:
Since odf-lvm versions are pinned like other MicroShift components, I would expect this to be part of the overall cluster upgrade workflow.  I.e. via an rpm-ostree layer

Reply:
I meant the developer task that would be part of our rebase process.

Reply:
Ah, right.  I'll add a blurb about this.  Really though, it should be implemented as part of the existing rebase process rather than be its own workflow, as it's essentially the same process as updating other core OCP image digests.